exporter/exporterhelper/queued_retry_test.go

```diff
@@ -582,6 +583,67 @@ func TestQueuedRetryPersistenceEnabledStorageError(t *testing.T) {
 	require.Error(t, be.Start(context.Background(), host), "could not get storage client")
 }
 
+func TestQueuedRetryPersistentEnabled_shutdown_dataIsRequeued(t *testing.T) {
+	produceCounter := atomic.NewUint32(0)
+
+	req := newMockRequest(context.Background(), 3, errors.New("some error"))
+	be := dataRequeuedTest(t, req, produceCounter)
+
+	// shuts down and ensure the item is produced in the queue again
+	require.NoError(t, be.Shutdown(context.Background()))
+	assert.Eventually(t, func() bool {
+		return produceCounter.Load() == uint32(2)
+	}, time.Second, 1*time.Millisecond)
+}
+
+func dataRequeuedTest(t *testing.T, req internal.Request, produceCounter *atomic.Uint32) *baseExporter {
+	storageID := component.NewIDWithName("file_storage", "storage")
+
+	qCfg := NewDefaultQueueSettings()
+	qCfg.NumConsumers = 1
+	qCfg.StorageID = &storageID // enable persistence
+	rCfg := NewDefaultRetrySettings()
+	rCfg.InitialInterval = time.Millisecond
+	rCfg.MaxElapsedTime = 0 // retry infinitely so shutdown can be triggered
+
+	be, err := newBaseExporter(defaultSettings, fromOptions(WithRetry(rCfg), WithQueue(qCfg)), "", nopRequestUnmarshaler())
+	require.NoError(t, err)
+
+	var extensions = map[component.ID]component.Component{
+		storageID: &mockStorageExtension{
+			mockClient: testutils.NewMockStorageClient(),
+		},
+	}
+	host := &mockHost{ext: extensions}
+
+	// we start correctly with a file storage extension
+	require.NoError(t, be.Start(context.Background(), host))
+
+	// wraps original queue so we can count operations
+	be.qrSender.queue = &producerConsumerQueueWithCounter{
+		ProducerConsumerQueue: be.qrSender.queue,
+		produceCounter:        produceCounter,
+	}
+
+	// replace nextSender inside retrySender to always return error so it doesn't exit send loop
+	castedSender, ok := be.qrSender.consumerSender.(*retrySender)
+	require.True(t, ok, "consumerSender should be a retrySender type")
+	castedSender.nextSender = &ErrorRequestSender{
+		errToReturn: errors.New("some error"),
+	}
+
+	go func() {
+		require.NoError(t, be.sender.send(req))
+	}()
```

Comment:
This could be potentially hazardous since if it tries to report an error once the test has finished, it can cause the test suite to panic. 

Ideally there should be a `sync.Waitgroup`, `chan` or `context` to help join the main test routine to spawned sub routines.